aopy/analysis.py

```diff
@@ -67,6 +71,39 @@ def factor_analysis_dimensionality_score(data_in, dimensions, nfold, maxiter=100
 
     return log_likelihood_score, iterations_required
 
+def get_pca_dimensions(data, max_dims=None, VAF=0.9, project_data=False):
+    """
+    Use PCA to estimate the dimensionality required to account for the variance in the given data. If requested it also projects the data onto those dimensions.
+    
+    Args:
+        data (nt, nch): time series data where each channel is considered a 'feature' (nt=n_samples, nch=n_features)
+        max_dims (int): (default None) the maximum number of dimensions
+                        if left unset, will equal the dimensions (number of columns) in the dataset
+        VAF (float): (default 0.9) variance accounted for (VAF)
+        project_data (bool): (default False). If the function should project the high dimensional input data onto the calculated number of dimensions
+
+    Returns:
+        tuple: Tuple containing: 
+            | **explained_variance (list):** variance accounted for by each principal component
+            | **num_dims (int):** number of principal components required to account for variance
+            | **projected_data (nt, ndims):** Data projected onto the dimensions required to explain the input variance fraction. If the input 'project_data=False', the function will return 'projected_data=None'
+    """
+    pca = PCA()
+    pca.fit(data)
+    explained_variance = pca.explained_variance_ratio_
+    total_explained_variance = np.cumsum(explained_variance)
+    if max_dims is None:
+        num_dims = np.min(np.where(total_explained_variance>VAF)[0])+1
+    else:
+        num_dims = max_dims
```

Comment:
seems to me like you would still want to do a comparison between the total_explained_variance, but just take the minimum of num_dims and max_dims

not sure. but this seems a bit pointless to just return the input argument if it's supplied